ensemble/randomforest.h

```diff
@@ -0,0 +1,53 @@
+#ifndef _MACHINELEARNING_ENSEMBLE_RANDOMFOREST_H_
+#define _MACHINELEARNING_ENSEMBLE_RANDOMFOREST_H_
+
+#include "../tree/tree.h"
+
+namespace ensemble {
+/**
+ * Base class for Random Forest Classification and Regression algorithms.
+ */
+class RandomForest {
+ public:
+  /**
+   * \brief Standard destructor.
+   */
+  virtual ~RandomForest();
+
+  /**
+   * Compute and return the score of the training dataset obtained using an
+   * out-of-bag estimate
+   * \return oob score
+   */
+  virtual double OobScore() = 0;
```

Comment:
Is this function returning the score for each sample in the Oob or just the coefficient of determination?

Since it is returning just a double value I am assuming that it is returning a metric value that represents the prediction for the entire set of samples in the Oob.

If this is the case, I think that we should also provide a function that returns the prediction for each sample in the Oob.